src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy

```diff
@@ -361,4 +361,30 @@ class PalantirDockerPluginTests extends AbstractPluginTest {
         execCond("docker rmi -f ${id}") || true
     }
 
+    def 'base image is pulled when "pull" parameter is set'() {
+        given:
+        String id = 'id8'
+        temporaryFolder.newFile('Dockerfile') << '''
+            FROM alpine:3.2
+        '''.stripIndent()
+        buildFile << """
+            plugins {
+                id 'com.palantir.docker'
+            }
+
+            docker {
+                name '${id}'
+                pull true
+            }
+        """.stripIndent()
+
+        when:
+        BuildResult buildResult = with('docker').build()
+
+        then:
+        buildResult.task(':docker').outcome == TaskOutcome.SUCCESS
+        buildResult.output.contains 'Pulling from library/alpine'
```

Comment:
You cannot rely on test ordering. The correct way is to first pull the base
image, then run the `docker` task and assert that it pulls again.

On Wed, Jul 27, 2016 at 10:59 AM Antoine Cotten <notifications@github.com>
wrote:

> In
> src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy
> <https://github.com/palantir/gradle-docker/pull/59#discussion_r72490130>:
>
> > +            plugins {
> > +                id 'com.palantir.docker'
> > +            }
> > +
> > +            docker {
> > +                name '${id}'
> > +                pull true
> > +            }
> > +        """.stripIndent()
> > +
> > +        when:
> > +        BuildResult buildResult = with('docker').build()
> > +
> > +        then:
> > +        buildResult.task(':docker').outcome == TaskOutcome.SUCCESS
> > +        buildResult.output.contains 'Pulling from library/alpine'
>
> I'm aware of that case, but the corresponding image will exist from
> previous tests. Should I maybe ensure that with an image dependency like
>
> dependencies {
>     docker 'alpine:3.2'
> }
>
> ?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/gradle-docker/pull/59/files/65657afc496e7eeb14eec5ffdb2c34c459b0407a#r72490130>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGOdwUV_GzYgu0pPPOf7ib86Ftx0_HGxks5qZ5xpgaJpZM4JWOh4>
> .
>